src/ILLink.Tasks/LinkTask.cs

```diff
@@ -102,42 +139,42 @@ public override bool Execute ()
 				var dir = Path.GetDirectoryName (assemblyPath);
 				if (!directories.Contains (dir)) {
 					directories.Add (dir);
-					args.Add ("-d");
-					args.Add (dir);
+					args.Append (" -d");
+					args.Append ($" {dir}");
```

Comment:
It's StringBuilder, you should avoid string concatenation (e.g. `args.Append(" ").Append(dir);` )